### PR TITLE
Default Value for Field in protocol

### PR DESCRIPTION
**Pull request by anandsubbu007**
Sorry for the unnecessary pull & push (I'm new to this).

some times it requires default value in SQL & in-class models. 

so, I modified the protocol YAML YAML file format to get the additional params to 'defaultvalue' to use this while creating SQL script & in-class generator.

### CODE REVIEW

**Comment by anandsubbu007** [comment]
made both old and this new format valid

Yaml file in protocol
```
fields:
  name: 
    type: String
    defaultvalue: 'A'
  listData: 
    type: List<String>
    defaultvalue: '[]'
  intType:
    type: int
    defaultvalue: 0
  emptyint: int
  emptyStrdata: String
```

SQL Syntax
```
CREATE TABLE example (
  "id" serial,
  "name" text NOT NULL DEFAULT 'A',
  "listData" json NOT NULL DEFAULT '[]',
  "intType" integer NOT NULL DEFAULT 0,
  "emptyint" integer NOT NULL,
  "emptyStrdata" text NOT NULL
);
```
In Class Model  constructor
```
  Example({
    this.id,
    this.name = 'A',
    this.listData = const [],
    this.intType = 0,
    required this.emptyint,
    required this.emptyStrdata,
  });
```

I belive this will be useful and need sometime

**Comment by vlidholt** [comment]
This looks like a really cool addition! I will need a few days to review it to make sure everything is correct (kind of important that everything works)

Before making large changes like this, it may be smart to clear it with me first, just to make sure that the updates fits into the grander plan of Serverpod. You can post suggestions on the discussion board here or message me on Linkedin: https://www.linkedin.com/in/viktorlidholt/

File: packages/serverpod_serialization/lib/src/serialization.dart [hunk 1]
```diff
@@ -83,6 +83,9 @@ abstract class SerializationManager {
         entity is double ||
         entity is SerializableEntity) {
       return '$entity';
+    } else if (entity is List<SerializableEntity>) {
+      // print('this is Seriziable List');
```
Comment (vlidholt) [inline packages/serverpod_serialization/lib/src/serialization.dart:87]: Avoid leaving empty comments in the code

Comment (anandsubbu007) [reply packages/serverpod_serialization/lib/src/serialization.dart:87]: ok

File: templates/serverpod_templates/modulename_server/lib/src/protocol/module_class.yaml [hunk 2]
```diff
@@ -5,5 +5,7 @@
 
 class: ModuleClass
 fields:
-  name: String
```
Comment (vlidholt) [inline templates/serverpod_templates/modulename_server/lib/src/protocol/module_class.yaml:8]: This is a breaking change for all Serverpod users. Also, it adds quite a bit of verbosity to the yaml files. The idea is to keep them as easy and clean as possible to read.

A solution would be to be able to both specify a field like:

```yaml
data: int
```

or

```yaml
data:
  type: int
  default: 32
```

**Comment by anandsubbu007** [comment]
> https://www.linkedin.com/in/viktorlidholt/

ok Sure

File: templates/serverpod_templates/projectname_flutter/android/local.properties [hunk 3]
```diff
@@ -0,0 +1,2 @@
+sdk.dir=/Users/apple/Library/Android/sdk
```
Comment (vlidholt) [inline templates/serverpod_templates/projectname_flutter/android/local.properties:1]: Avoid committing files that aren't part of the actual feature you are working on, this makes it much harder to review.

File: tools/serverpod_cli/bin/generator/pgsql_generator.dart [hunk 4]
```diff
@@ -34,19 +34,27 @@ class PgsqlGenerator {
     out += '--\n';
     out += '\n';
 
+    // Drop Table If Already Exist
```
Comment (vlidholt) [inline tools/serverpod_cli/bin/generator/pgsql_generator.dart:37]: Why was this commented out?

File: templates/serverpod_templates/projectname_flutter/android/local.properties (hunk 3 above)
Comment (anandsubbu007) [reply templates/serverpod_templates/projectname_flutter/android/local.properties:1]: ok, sorry about that. i didn't check it before pushing it

File: templates/serverpod_templates/modulename_server/lib/src/protocol/module_class.yaml (hunk 2 above)
Comment (anandsubbu007) [reply templates/serverpod_templates/modulename_server/lib/src/protocol/module_class.yaml:8]: both works

Comment (anandsubbu007) [reply templates/serverpod_templates/modulename_server/lib/src/protocol/module_class.yaml:8]: i checked for its type that it is string or map and do func based on its type.
i belive this format will also help us to extend some column property  that we may define in sql like. primary key, foreignkey

File: tools/serverpod_cli/bin/generator/pgsql_generator.dart (hunk 4 above)
Comment (anandsubbu007) [reply tools/serverpod_cli/bin/generator/pgsql_generator.dart:37]: i want to implement this function. 
while development. i want to change table column & its properties. 
i'm looking for sql script to alter existing column.
i looked for script. but didn't get it.
tring to take a copy of old existing table and drop it and recreate it and then copy old data to newly created table and then to drop copy table.

still didn't know the script to copy existing data to new data which match column name without losing it.

**Comment by anandsubbu007** [comment]
added some extension to ser
```
  Future<ExampleClass?> getById(Session session, int id) async {
    final data = await ExampleClass.findById(session, id);
    data?.resStatusCode = 200;
    data?.resContent = 'This is sample Content The will Sent Additionally';
    data?.resAdditionalValue = true;
return data;
```

on receiver side
```
      final data = await client.exampleFunc.getById(id);
      print('Testing With Response');
      print('statusCode: ${data.resStatusCode}');
      print('Content: ${data.resContent}');
      print('AddedVal: ${data.resAdditionalValue}');
```

File: tools/serverpod_cli/bin/generator/class_generator_dart.dart [hunk 5]
```diff
@@ -574,23 +584,38 @@ class FieldDefinition {
...
+    if (description is Map) {
+      _typeStr = description['type'];
+      var _scopeStr = description['scope'];
+      if (_scopeStr == 'database') {
         scope = FieldScope.database;
-      } else if (scopeStr == 'api') {
+      } else if (_scopeStr == 'api') {
         scope = FieldScope.api;
       }
+      // if (['String', 'int', 'double', 'bool','DateTime'].contains(typeStr)) {
```
Comment (vlidholt) [inline tools/serverpod_cli/bin/generator/class_generator_dart.dart:597]: Avoid leaving comments in there

File: tests/serverpod_test_server/lib/src/protocol/types.yaml [hunk 6]
```diff
@@ -1,9 +1,19 @@
 class: Types
```
Comment (vlidholt) [inline tests/serverpod_test_server/lib/src/protocol/types.yaml:1]: Please make sure that no changes are made to these files

File: tools/serverpod_cli/bin/generator/class_generator_dart.dart (hunk 5 above)
Comment (anandsubbu007) [reply tools/serverpod_cli/bin/generator/class_generator_dart.dart:597]: removed this in other pr

File: tests/serverpod_test_server/lib/src/protocol/types.yaml (hunk 6 above)
Comment (anandsubbu007) [reply tests/serverpod_test_server/lib/src/protocol/types.yaml:1]: ok i think that it will help when testing

Comment (anandsubbu007) [reply tests/serverpod_test_server/lib/src/protocol/types.yaml:1]: i replaced all data in this test folder as before in https://github.com/serverpod/serverpod/pull/194

**Comment by anandsubbu007** [comment]
added this in the following pr

**Pull request closed without merging**